src/middlewared/middlewared/plugins/reporting/events.py

```diff
@@ -63,7 +63,16 @@ def run_sync(self):
 
         while not self._cancel_sync.is_set():
             # this gathers the most recent metric recorded via netdata (for all charts)
-            netdata_metrics = self.middleware.call_sync('netdata.get_all_metrics')
+            retries = 2
+            while retries > 0:
+                try:
+                    netdata_metrics = self.middleware.call_sync('netdata.get_all_metrics')
+                except Exception:
+                    retries -= 1
```

Comment:
We should sleep an arbitrary amount of time instead of trying as fast as a cpu core allows.